functions/index.js

```diff
@@ -34,28 +40,61 @@ exports.createNewTask = functions.https.onRequest((request, response) => {
-                Title: title,
-                Description: des,
-                Priority: priority,
-                Difficulty: difficulty,
-                Creator: creator,
-                Assignee: assignee,
-                ET: estimatedTime,
-                Status: status,
-                Category: category,
-                LogWorkTotalTime: loggedWorkTotalTime,
-                WorkDone: workDone,
-                CreateNewTaskSprintNumber: createNewTaskSprintNumber
+        db.collection("Main").doc("RawData").get().then((doc) => {
+
+                totalDevelopmentTask = doc.data().TotalDevelopmentTask + 1;
+                totalBusinessTask = doc.data().TotalBusinessTask + 1;
+                totalMarketingTask = doc.data().TotalBusinessTask + 1;
+                totalNumberOfTask = (totalDevelopmentTask + totalBusinessTask + totalMarketingTask);
+
+                if (category === "Development") {
+                    taskId = category[0] + totalDevelopmentTask;
+                } else if (category === "Business") {
+                    taskId = category[0] + totalBusinessTask;
+                } else {
+                    taskId = category[0] + totalMarketingTask;
+                }
+
+                console.log(taskId);
+
+                var setDataPromise = db.collection(fullSprintId).doc(taskId).set({
+                    Title: title,
+                    Description: des,
+                    Priority: priority,
+                    Difficulty: difficulty,
+                    Creator: creator,
+                    Assignee: assignee,
+                    ET: estimatedTime,
+                    Status: status,
+                    Category: category,
+                    LogWorkTotalTime: loggedWorkTotalTime,
+                    WorkDone: workDone,
+                    CreateNewTaskSprintNumber: createNewTaskSprintNumber,
+                    StoryPointNumber: storyPointNumber
+                });
+                return Promise.resolve(setDataPromise);
             })
-            .then(() => {
-                var work = { data: "working" }
+            .then(function(setDataPromise) {
+                var updateSetDataPromise = db.collection("Main").doc("RawData").update({
+                    TotalDevelopmentTask: totalDevelopmentTask,
+                    TotalBusinessTask: totalBusinessTask,
+                    TotalMarketingTask: totalMarketingTask,
+                    TotalNumberOfTask: totalNumberOfTask
+
+                });
+                return Promise.resolve(updateSetDataPromise);
+            })
+            .then((updateSetDataPromise) => {
+                var result = { data: "working" }
                 console.log("Document successfully written!");
-                return response.status(200).send(work);
+                return response.status(200).send(result);
             })
             .catch(() => {
+                var error = "not Working";
```

Comment:
Please catch the error in catch block.

And in replace var error with result;

example:

.catch((error) => {
   result = { data: error }
}

And also create the variable result as out of these block.

example:
var result = {data: "OK"};

Note: Please don't make it global but keep it out of the scope of .then() and .catch()

Reply:
okay i have made changes